### DISCUSSION TITLE
Is it possible to explore the content of local_storage?

### DISCUSSION THREAD

**Discussion by stenci** [anvilistas/anvil-extras#296] [General]
**Question 1**

Is it possible to have a look at the whole content of `local_storage` for an app, so I have an idea if all is working as expected?

---

**Question 2 / Request**

Do you think it is possible to _bind_ the value of a component to a value in `local_storage`?

I would like to do something like this in the `__init__` of a form:
```
#                            component        property   default value
local_storage.bind_component(self.text_box_1, 'text',   'hello')
```
and have `self.text_box_1.text` populated when the form is created and its value saved to local storage whenever it changes. Perhaps the values should be saved when the form is hidden... but this would fail to save if the user closes the page without hiding the form. Perhaps they should be saved as soon as they are changed... I don't really know what would be the best behavior.

Thoughts?

---

**Use case**

All my forms have this in `__init__`:
```
self.panels_page_size.selected_value = local_storage.get('DefineCrates.panels_page_size', '20 rows')
```
and this called when any of the component values changes:
```
def update_local_storage(self):
    local_storage['DefineCrates.panels_page_size'] = self.panels_page_size.selected_value
```

Some forms have 20-30 of these lines, one for each value I want to remember, for each checkbox, dropdown, etc. in the form. The whole app can have hundreds of items stored in `local_storage`.

Having `local_storage.bind_component` would still require the 20-30 lines in `__init__`, but would not require the definition of `def update_local_storage(self):` on each form and calling it whenever some component value changes.

These snippets come from the `DefineCrates` form that has 5 DataGrids, one of them called `panels_page` that comes with a drop down called `panels_page_size`. Here you can see 9 of the values stored in `local_storage`, there are more in the hidden _tabs_.
![image](https://user-images.githubusercontent.com/5955495/159989504-ff2d1a7e-4197-4e09-9356-5c525e3e659a.png)

**Comment by meatballs** [comment]
1. Yes - in the developer tools for your browser, you should find a 'Storage' tab which will let you browse the content of both local storage and Indexed DB
2. That's the sort of thing that the [atomic](https://github.com/anvilistas/anvil-labs/blob/main/docs/guides/modules/atomic.rst) module in anvil labs will enable.
